### PR TITLE
Add ethernet:interface_mode setting to yaml [ESD-1320]

### PR DESCRIPTION
**Pull request by yizow**
Add description for new ethernet:interface_mode setting

### CODE REVIEW

File: piksi_tools/console/settings.yaml [hunk 1]
```diff
@@ -1701,6 +1701,20 @@
   Description: Enable the almanac-based acquisition.
   Notes:
 
+- group: ethernet
+  name: mode
+  expert: false
+  type: boolean
+  enumerated possible values: True,False
```
Comment (benjaminaltieri) [inline piksi_tools/console/settings.yaml]: type should be enum, values `active` or `config`

File: piksi_tools/console/settings.yaml [hunk 2]
```diff
@@ -1701,6 +1701,20 @@
...
+  units: N/A
+  default value: 'False'
```
Comment (benjaminaltieri) [inline piksi_tools/console/settings.yaml]: needs update

File: piksi_tools/console/settings.yaml [hunk 3]
```diff
@@ -1701,6 +1701,20 @@
...
+  readonly: false
+  Description: Ethernet configuration locked toggle.
```
Comment (benjaminaltieri) [inline piksi_tools/console/settings.yaml]: "Ethernet configuration mode:"

File: piksi_tools/console/settings.yaml [hunk 4]
```diff
@@ -1701,6 +1701,20 @@
...
+  Notes: |
+    "Config" IP configuration can be changed freely, but no change is made on the device.
```
Comment (benjaminaltieri) [inline piksi_tools/console/settings.yaml]: Append "Changing setting to Active will refresh ethernet connection with current values."

Comment (denniszollo) [reply piksi_tools/console/settings.yaml]: I can't determine how this setting works from the description.  If in "active" mode, does the device behave as it used to on prior releases?

Comment (benjaminaltieri) [reply piksi_tools/console/settings.yaml]: Underlying behavior is that we want 'active' to denote currently active settings and delay application of settings changes to individual settings until it is desired to do so. The solution is to change to 'config' mode which allows you to change the setting without them taking effect. Switching back to 'active' will apply the settings and re-up the interface in one go

File: piksi_tools/console/settings.yaml [hunk 5]
```diff
@@ -1701,6 +1701,20 @@
...
+  name: interface_mode
+  expert: false
+  type: enum
+  enumerated possible values: Config,Active
+  units: N/A
+  default value: 'Active'
+  readonly: false
+  Description: Ethernet configuration mode.
+  Notes: |
+    "Config" IP configuration can be changed freely, but no change is made on the device. Returning to 'active' mode will refresh ethernet connection with current values.
```
Comment (silverjam) [inline piksi_tools/console/settings.yaml]:
```suggestion
    "Config" IP configuration can be changed freely, but no change is made on the device. Returning to 'Active' mode will refresh ethernet connection with current values.
```

File: piksi_tools/console/settings.yaml [hunk 6]
```diff
@@ -1701,6 +1701,20 @@
...
+
+    "Active" The current IP configuration is sent to the device and updated. Afterward, no IP settings can be changed until returned to 'config' mode.
```
Comment (silverjam) [inline piksi_tools/console/settings.yaml]:
```suggestion
    "Active" The current IP configuration is sent to the device and updated. Afterward, no IP settings can be changed until returned to 'Config' mode.
```

**Pull request merged**